src/components/Tabs/Tabs.tsx

```diff
@@ -28,70 +29,84 @@ export interface TabsProps extends QAProps {
     activeTab?: string;
     /** By default if activeTab is not set, first tab will be active */
     allowNotSelected?: boolean;
-    /** Tabs */
-    items: TabsItemProps[];
+    /** Tabs props list */
+    items?: TabsItemProps[];
+    children?: React.ReactNode;
     /** Additional CSS-class */
     className?: string;
     /** Select tab handler */
     onSelectTab?(tabId: string): void;
-    /** Allows to wrap `TabItem` into another component or render custom tab  */
+    /** Allows to wrap `TabItem` into another component or render custom tab.
+     * Ignored when tabs rendered via `children` */
     wrapTo?(item: TabsItemProps, node: React.ReactNode, index: number): React.ReactNode;
 }
 
-export const Tabs = forwardRef<HTMLDivElement, TabsProps>(
+const getActiveTabId = (
+    activeTab: TabsProps['activeTab'],
+    allowNotSelected: TabsProps['allowNotSelected'],
+    items: TabsProps['items'],
+) => {
+    if (activeTab) {
+        return activeTab;
+    }
+
+    if (allowNotSelected || items?.length === 0) {
```

Comment:
We need a the second condition here?

Reply:
I have moved this code as is, since we don't need to memoize string